<?xml version='1.0' encoding='UTF-8'?>
<!--Generated automatically from launch config file.-->
<launch xmlns:xacro="http://www.ros.org/wiki/xacro">
  <xacro:property name="params" value="${load_yaml('../../config/.launch_params.yaml')}"/>
  <xacro:property name="method" value="${params['past_image']['general']['eval_method']}"/>
  <xacro:property name="method_ns" value="${params['past_image'][method]}"/>
  <node name="past_image_selector" output="screen" pkg="spirit" type="past_image_selector.py">
    <param name="eval_method" value="${method}"/>
    <param name="image_queue_length" value="${params['past_image']['general']['image_queue_length']}"/>
    <xacro:if value="${method == 'ConstantTimeDelay'}">
      <param name="ref_delay" value="${method_ns['ref_delay']}"/>
      <param name="coeff_time" value="${method_ns['coeff_time']}"/>
    </xacro:if>
    <xacro:if value="${method == 'Spirit'}">
      <param name="coeff_centrality" value="${method_ns['coeff_centrality']}"/>
      <param name="coeff_distance_with_current" value="${method_ns['coeff_distance_with_current']}"/>
      <param name="coeff_direction_with_current" value="${method_ns['coeff_direction_with_current']}"/>
      <param name="thresh_yaw" value="${3.14159/180*method_ns['thresh_yaw']}"/>
      <param name="coeff_centrality2" value="${method_ns['coeff_centrality2']}"/>
      <param name="coeff_distance" value="${method_ns['coeff_distance']}"/>
      <param name="ref_distance" value="${method_ns['ref_distance']}"/>
      <param name="coeff_direction" value="${method_ns['coeff_direction']}"/>
      <param name="thresh_distance" value="${method_ns['thresh_distance']}"/>
    </xacro:if>
    <xacro:if value="${method == 'Murata'}">
      <param name="coeff_distance" value="${method_ns['coeff_distance']}"/>
      <param name="ref_distance" value="${method_ns['ref_distance']}"/>
      <param name="coeff_height" value="${method_ns['coeff_height']}"/>
      <param name="ref_height" value="${method_ns['ref_height']}"/>
      <param name="coeff_direction" value="${method_ns['coeff_direction']}"/>
      <param name="coeff_elevation" value="${method_ns['coeff_elevation']}"/>
    </xacro:if>
    <xacro:if value="${method == 'ConstantDistance'}">
      <param name="coeff_distance" value="${method_ns['coeff_distance']}"/>
      <param name="ref_distance" value="${method_ns['ref_distance']}"/>
    </xacro:if>
  </node>
</launch>
